Ncto 135 make an option to share the game using a link

Louis-rollet opened the pull request:
## Jira ticket  
https://naucto.atlassian.net/browse/NCTO-66
https://naucto.atlassian.net/browse/NCTO-135

## What does your MR do ?  
Added in the project settings a button to publish the project that then will be accessible in the hub part which contains for now all published projects. When clicking on a published game on the hub it will open a window which contains the game and its description

## How to test it  
create a project and publish it, then test it by going to the hub

## Screenshot  
<img width="987" height="926" alt="image" src="https://github.com/user-attachments/assets/aa688f0c-0cac-4796-b0ab-70a68227a347" />
<img width="1802" height="954" alt="image" src="https://github.com/user-attachments/assets/db7b051c-b917-4a87-964d-dbb26cdee7d3" />



## Notes

Louis-rollet committed:
[HUB] [ADD] published state to games and hub game lisiting + playing window with project desc and name

Louis-rollet committed:
[HUB] [FIX] remove getcanva function

Louis-rollet committed:
[GAME] [FIX] corner wrong round corner

Louis-rollet committed:
Merge branch 'main' of github.com:Naucto/Frontend into NCTO-135-make-an-option-to-share-the-game-using-a-link

Louis-rollet committed:
[HUB] [FIX] now using the new bakcend api for the publish and hub loading of published games

Louis-rollet committed:
[PROJECT] [FIX] minor changes on the serivce and the gameprovier

Louis-rollet committed:
[CDN][ADD] handling of cdn urls if available for the published games

Louis-rollet committed:
Merge branch 'main' of github.com:Naucto/Frontend into NCTO-135-make-an-option-to-share-the-game-using-a-link

alexis-belmonte commented on src/modules/hub/Hub.tsx:
```diff
@@ -1,12 +1,177 @@
+  margin: theme.spacing(4),
 }));
 
+const CategorySection = styled(Box)(({ theme }) => ({
```
All these things could be in a separate generic tsx file, but this can be done in a separate PR --> `src/components/ui`

alexis-belmonte commented on src/modules/create/game-editor/editors/ProjectSettingsEditor.tsx:
```diff
@@ -1,8 +1,8 @@
+import { Box, Button, Typography, List, ListItem, ListItemText, Paper, Divider, Chip } from "@mui/material";
 import { styled } from "@mui/material/styles";
-import { ApiError, ProjectsService, ProjectWithRelationsResponseDto, UserBasicInfoDto } from "@api";
+import { ApiError, ProjectsService, ProjectWithRelationsResponseDto, UserBasicInfoDto, } from "@api";
```
Unnecessary change

Louis-rollet replied:
yes

alexis-belmonte commented on src/modules/hub/components/GameViewer.tsx:
```diff
@@ -0,0 +1,207 @@
+import { EnvData } from "@shared/luaEnvManager/LuaEnvironmentManager";
+import { GameProvider, ProviderEventType } from "@providers/GameProvider";
+
+const Overlay = styled(Box)(() => ({
```
If possible, move those components to src/ui/components as well -- but only if those components are generic enough

alexis-belmonte commented on src/modules/create/game-editor/editors/ProjectSettingsEditor.tsx:
```diff
@@ -84,10 +93,45 @@ const ProjectSettingsEditor: React.FC<EditorProps> = ({ project }) => {
+      setIsPublished(!isPublished);
+    } catch (err) {
+      alert("Error updating publish status: " +
+        (err instanceof ApiError ? err.message : String(err)));
```
Add a TODO here

Louis-rollet replied:
[f9046b2](https://github.com/Naucto/Frontend/pull/43/commits/f9046b261739ad23002e5c10a836e071ce174ae5)

alexis-belmonte commented on src/modules/create/game-editor/editors/ProjectSettingsEditor.tsx:
```diff
@@ -23,16 +23,25 @@ const CollaboratorList = styled(List)(({ theme }) => ({
   borderRadius: theme.shape.borderRadius,
 }));
 
+const StatusContainer = styled(Box)(({ theme }) => ({
```
If possible, move those components to src/ui/components as well -- but only if those components are generic enough

alexis-belmonte commented on src/modules/projects/components/ProjectCard.tsx:
```diff
@@ -38,10 +39,14 @@ const ProjectFooter = styled("div")(({ theme }) => ({
+      navigate(urls.toProjectView(project.id));
+    } else {
+      navigate(urls.toProject(project.id));
+    };
```
True

alexis-belmonte reviewed (changes requested):
好极了！

Notgoyome commented on src/modules/editor/SpriteEditor/SpriteEditor.css:
question: why is there this css file ?

Notgoyome commented on src/modules/hub/components/GameViewer.tsx:
```diff
@@ -0,0 +1,207 @@
+export const GameViewer: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+  const navigate = useNavigate();
+  const [project, setProject] = useState<ProjectWithRelationsResponseDto | null>(null);
```
question: undefined not possible ?

Louis-rollet committed:
[API][UPDATE] change CloudFront references to Edge in SignedCdnResourceDto and SignedUrlResponseDto; enhance URL handling in GameProvider for CORS

Louis-rollet committed:
[GAME PROVIDER][UPDATE] enhance error handling with ReleaseContentFetchError; improve cleanup in destroy method
[PROJECT SETTINGS][FIX] correct logical operator in setIsPublished; add TODO for publish/unpublish behavior clarification
[PROJECT CARD][FIX] fix syntax error by removing unnecessary semicolon
[REQUEST][UPDATE] mark generated file as editable

Louis-rollet committed:
[CODE][UPDATE] format import statements for consistency

Louis-rollet commented on src/modules/create/game-editor/editors/ProjectSettingsEditor.tsx:
```diff
@@ -84,10 +93,45 @@ const ProjectSettingsEditor: React.FC<EditorProps> = ({ project }) => {
+      alert("Error updating publish status: " +
+        (err instanceof ApiError ? err.message : String(err)));
+    }
+  };
```
[f9046b2](https://github.com/Naucto/Frontend/pull/43/commits/f9046b261739ad23002e5c10a836e071ce174ae5)

Louis-rollet committed:
[GAME VIEWER][UPDATE] change project state initialization from null to undefined for better type handling

Louis-rollet commented on src/providers/GameProvider.ts:
```diff
@@ -0,0 +1,94 @@
+        try {
+          const response = await fetch(signed.signedUrl);
+          if (!response.ok) {
+            throw new Error(`Failed to fetch release content: ${response.status}`);
```
[f9046b2](https://github.com/Naucto/Frontend/pull/43/commits/f9046b261739ad23002e5c10a836e071ce174ae5)

Louis-rollet committed:
Merge branch 'main' of github.com:Naucto/Frontend into NCTO-135-make-an-option-to-share-the-game-using-a-link

alexis-belmonte reviewed (approved):
![](https://i.ebayimg.com/images/g/FGMAAOSwBvNTpwXp/s-l1200.jpg)

Louis-rollet committed:
[SPRITE EDITOR][REMOVE] delete unused CSS styles from SpriteEditor.css
